src/main/java/core/basesyntax/figures/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax.figures;
+
+import java.util.Locale;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double topSide;
+    private double bottomSide;
+    private double lateralSide;
+    private double area;
+
+    public IsoscelesTrapezoid(String color, double topSide, double bottomSide, double lateralSide) {
+        super(color);
+        this.topSide = topSide;
+        this.bottomSide = bottomSide;
+        this.lateralSide = lateralSide;
+        this.area = calculateArea();
+    }
+
+    public double getTopSide() {
+        return topSide;
+    }
+
+    public void setTopSide(double topSide) {
+        this.topSide = topSide;
+        this.area = calculateArea();
+    }
+
+    public double getBottomSide() {
+        return bottomSide;
+    }
+
+    public void setBottomSide(double bottomSide) {
+        this.bottomSide = bottomSide;
+        this.area = calculateArea();
+    }
+
+    public double getLateralSide() {
+        return lateralSide;
+    }
+
+    public void setLateralSide(double lateralSide) {
+        this.lateralSide = lateralSide;
+        this.area = calculateArea();
+    }
+
+    public double getArea() {
+        return area;
```

Comment:
If the area is calculated on-demand, this getter could call `calculateArea()` directly, and there would be no need to store 'area' as a field.